src/epplibwrapper/client.py

```diff
@@ -60,44 +64,99 @@ def __init__(self) -> None:
                 password=settings.SECRET_REGISTRY_KEY_PASSPHRASE,
             )
         )
-        # prepare a context manager which will connect and login when invoked
-        # (it will also logout and disconnect when the context manager exits)
-        self._connect = Socket(self._client, self._login)
+
+        self.pool_options = {
+            # Pool size
+            "size": settings.EPP_CONNECTION_POOL_SIZE,
+            # Which errors the pool should look out for.
+            # Avoid changing this unless necessary,
+            # it can and will break things.
+            "exc_classes": (TransportError,),
+            # Occasionally pings the registry to keep the connection alive.
+            # Value in seconds => (keepalive / size)
+            "keepalive": settings.POOL_KEEP_ALIVE,
+        }
+
+        self._pool = None
+
+        # Tracks the status of the pool
+        self.pool_status = PoolStatus()
+
+        if start_connection_pool:
+            self.start_connection_pool()
 
     def _send(self, command):
         """Helper function used by `send`."""
+        cmd_type = command.__class__.__name__
+
+        # Start a timeout to check if the pool is hanging
+        timeout = Timeout(settings.POOL_TIMEOUT)
+        timeout.start()
+
         try:
-            cmd_type = command.__class__.__name__
-            with self._connect as wire:
-                response = wire.send(command)
+            if not self.pool_status.connection_success:
+                raise LoginError(
+                    "Couldn't connect to the registry after three attempts"
+                )
+            with self._pool.get() as connection:
+                response = connection.send(command)
+        except Timeout as t:
+            # If more than one pool exists,
+            # multiple timeouts can be floating around.
+            # We need to be specific as to which we are targeting.
+            if t is timeout:
```

Comment:
Yup! Basically, lets say that we are running three pools at once. Two of them are dead, but the third is running fine. gevent.timeouts are _global_ so if this line doesn't exist, it will kill all three pools rather than the dead ones.
https://www.gevent.org/api/gevent.timeout.html
![image](https://github.com/cisagov/getgov/assets/141044360/12b3190b-ee5c-4a5f-979f-938e75373c36)